csrc/multidevice/utils.cpp

```diff
@@ -234,8 +236,22 @@ std::vector<int64_t> unshardedSizes(
     const TensorView* tv,
     c10::IntArrayRef sizes) {
   std::vector<int64_t> unsharded_sizes = sizes.vec();
+
+  // Ideally, the unsharded sizes should be inferred from the
+  // allocation domain. The allocation domain for multidevice
```

Comment:
```suggestion
  // The allocation domain for multidevice
```
I wouldn't say "ideally". We chose the contract that allocation is finalized during preseg. We think it's the best choice at this moment; otherwise PropagateShardingsPass would have to propagate both loop and allocation. What unshardedSizes does here implements that contract and there's nothing wrong for that.